### ISSUE TITLE
avoid_unused_constructor_parameters: Doesn't understand redirects

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/linter#863]
Valid, but causes a lint:

```dart
abstract class A {
  factory A(String name) = B;
}

class B {
  final String name;

  B(this.name);
}
```

**Comment by a14n** [comment]
What's your linter version ? I fixed it in #847 released in _0.1.40_.

**Comment by pq** [comment]
Does `B` need to implement `A` in your example?

As @a14n pointed out, this appears to work as of `0.1.40`; likely your SDK doesn't have that rolled in?  (FWIW, you'll want one post https://github.com/dart-lang/sdk/commit/8f84f3c890a7fbbeea7e802215bac13323edf36f#diff-946813de8d9b30857edae58aec4f3a7f.)

**Comment by pq** [comment]
Fixed in `0.1.40`.
